src/App.tsx

```diff
@@ -1,27 +1,80 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import 'bulma/css/bulma.min.css';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
+import { LoadingError } from './components/LoadingError/LoadingError';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good, GoodsType, LoadError } from './types/Good';
+import { Loader } from './components/Loader/Loader';
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+  const [currentGoodsType, setCurrentGoodsType] = useState(GoodsType.Default);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const loadGoods = (
+    getGoods: () => Promise<Good[]>,
+    goodsType = GoodsType.Default,
+  ) => {
+    if (goodsType === currentGoodsType) {
+      return;
+    }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    setLoading(true);
```

Comment:
- Clear an error
- Don't show the list if an error


https://user-images.githubusercontent.com/54940595/230579093-2275e3a2-a7fe-4f13-ac33-0dd9e61cc7ea.mov